ducktape/tests/loader.py

```diff
@@ -55,10 +55,13 @@ class TestLoader(object):
 
     def __init__(self, session_context):
         self.session_context = session_context
-        self.logger = session_context.logger
         self.test_file_pattern = DEFAULT_TEST_FILE_PATTERN
         self.test_function_pattern = DEFAULT_TEST_FUNCTION_PATTERN
 
+    @property
```

Comment:
Why are all these fields being converted into property methods? Did something change that made them dynamic such that we need to look them up every time?